src/arrayMethodSort.js

```diff
@@ -4,8 +4,30 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i; j < this.length; j++) {
+        if (compareFunction) {
+          if (compareFunction(this[i], this[j]) > 0) {
+            const x = this[i];
+
+            this[i] = this[j];
+            this[j] = x;
+          }
+        } else {
+          if (String(this[j]) < String(this[i])) {
+            const x = this[i];
+
+            this[i] = this[j];
+            this[j] = x;
+          }
+        }
+      }
+    }
```

Comment:
The bubble sort algorithm can be optimized by recognizing that after each pass, the largest element is bubbled to the end. Thus, the inner loop can be reduced by 'i' since the last 'i' elements are already sorted.